patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1175,7 +1175,7 @@
  				cloudBGActive = 0f;
  
 -			if (cloudBGActive == 0f && rand.Next((int)((float)(num2 * 12 / ((dayRate == 0) ? 1 : dayRate)) / num3)) == 0) {
-+			if (cloudBGActive == 0f && rand.Next(((num2 * 12 / ((desiredWorldEventsUpdateRate == 0) ? 1 : desiredWorldEventsUpdateRate)) / num3)) == 0) {
++			if (cloudBGActive == 0f && rand.NextDouble() <= 1 / ((num2 * 12 / desiredWorldEventsUpdateRate) / num3)) {
```

Comment:
change
`1 / ((num2 * 12 / desiredWorldEventsUpdateRate) / num3)`
to
`num3 * desiredWorldEventsUpdateRate / (num2 * 12 )`